dnf-docker-test/features/steps/groups_steps.py

```diff
@@ -0,0 +1,130 @@
+<!DOCTYPE comps PUBLIC "-//Red Hat, Inc.//DTD Comps info//EN" "comps.dtd">
+<comps>
+"""
+COMPS_TMPL = """
+  <group>
+   <id>{{ name }}</id>
+   <default>{{ is_default|default("false") }}</default>
+   <uservisible>{{ is_uservisible|default("true") }}</uservisible>
+   <display_order>1024</display_order>
+   <name>{{ name }}</name>
+   <description>{{ description|default("") }}</description>
+    <packagelist>
+{%- if mandatory is defined %}
+{% for pkg in mandatory %}
+      <packagereq type="mandatory">{{ pkg }}</packagereq>
+{%- endfor %}
+{%- endif %}
+{%- if default is defined %}
+{% for pkg in default %}
+      <packagereq type="default">{{ pkg }}</packagereq>
+{%- endfor %}
+{%- endif %}
+{%- if optional is defined %}
+{% for pkg in optional %}
+      <packagereq type="default">{{ pkg }}</packagereq>
+{%- endfor %}
+{%- endif %}
+    </packagelist>
+  </group>
+"""
+COMPS_SUFFIX = """</comps>"""
+
+REPO_TMPL = "/etc/yum.repos.d/{!s}.repo"
+
+JINJA_ENV = jinja2.Environment(undefined=jinja2.StrictUndefined)
+
+
+@given('package groups defined in repository "{repository}"')
+def given_package_groups_defined_in_repository(ctx, repository):
+    """
+    For a given repository creates comps.xml file with described
+    package groups and recreates the repo
+
+    .. note::
+
+       Requires *createrepo_c* and the repo to be already created.
+
+    Requires table with following headers:
+
+    ========= ===== =======
+     Group     Tag   Value
+    ========= ===== =======
+
+    *Tag* is describing characteristics of the respective
+    package group.Supported tags are:
+
+    ============== ===============
+         Tag        Default value 
+    ============== ===============
+    is_default          false     
+    is_uservisible      true      
+    description         ""        
+    mandatory           []        
+    default             []        
+    optional            []        
+    ============== ===============
+
+    Examples:
+
+    .. code-block:: gherkin
+
+       Feature: Installing a package group
+
+         @setup
+         Scenario: Repository base with package group minimal
+              Given repository "base" with packages
+                 | Package | Tag | Value |
+                 | foo     |     |       |
+                 | bar     |     |       |
+              And package groups defined in repository "base"
+                 | Group    | Tag        | Value |
+                 | minimal  | mandatory  | foo   |
+                 |          | default    | bar   |
+
+         Scenario: Installing package group from background
+              When I enable repository "base"
+              Then I successfully run "dnf -y group install minimal"
+    """
+    HEADINGS_GROUP = ['Group', 'Tag', 'Value']
+    GROUP_TAGS_REPEATING = ['mandatory', 'default', 'optional']
+    GROUP_TAGS = ['is_default', 'is_uservisible', 'description'] + \
+                 GROUP_TAGS_REPEATING
```

Comment:
What is the preferred way of fixing this? Should I indent only using 4 spaces or should I rather use one long line?